subsys/bluetooth/audio/bap_scan_delegator.c

```diff
@@ -535,7 +547,23 @@ static int scan_delegator_add_source(struct bt_conn *conn,
 		return BT_GATT_ERR(BT_ATT_ERR_VALUE_NOT_ALLOWED);
 	}
 
-	state->broadcast_id = net_buf_simple_pull_le24(buf);
+	broadcast_id = net_buf_simple_pull_le24(buf);
+
+	/* Check whether this add_source operation would result in a duplication */
+	lookup_state = bass_lookup_broadcast_id(broadcast_id);
+	if (lookup_state != NULL && lookup_state->state.adv_sid == state->adv_sid
+	    && lookup_state->state.addr.type == state->addr.type) {
+		char addr_str[BT_ADDR_LE_STR_LEN];
+
+		bt_addr_le_to_str(bt_conn_get_dst(conn), addr_str, sizeof(addr_str));
+		LOG_DBG("recv_state with src_id=0x%02X already uses broadcast_id=0x%06X, "
+			"adv_sid=0x%02X, and addr.type=0x%02X",	lookup_state->state.src_id,
+			broadcast_id, lookup_state->state.adv_sid, lookup_state->state.addr.type);
+
+		return BT_GATT_ERR(BT_ATT_ERR_WRITE_REQ_REJECTED);
+	}
```

Comment:
Consider moving this into a `bool bass_is_duplicate(...)` function. 

Does the BAP spec say anything about `modify source`? Do we want to prevent any source from being modified with duplicate values as well?

Reply:
> Consider moving this into a `bool bass_is_duplicate(...)` function.

Wil do


> Does the BAP spec say anything about `modify source`? Do we want to prevent any source from being modified with duplicate values as well?


It does not, but the Modify Source operation does not allow any of three values using in duplication critera to be changed. It only allows changing `PA_Sync`, `PA_Interval`, `Num_Subgroups`, `BIS_Sync`, and applying metadata.

Reply:
Ah, you are right :)